### ISSUE TITLE
`Badge`: Add property to toggle truncation

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#68091]
This came up while trying to tackle https://github.com/WordPress/gutenberg/issues/68020. 

The anchor 'badge' in List View is truncated to prevent it from taking over the entire row:

<img width="349" alt="Image" src="https://github.com/user-attachments/assets/c528db10-1aaf-4dec-8d63-f929dd5e288c" />

In order to replace this ad hoc element with the `Badge` component, it would need to be updated to support truncation.

The [`Text`](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-text--docs) component already supports truncation, so perhaps this is just a case of utilising `Text` inside `Badge`, and providing a way to pass the `truncate` property down to `Text`.

cc @WordPress/gutenberg-components

**Comment by mirka** [comment]
There are some issues with the `Truncate` API, so I'm considering adding direct truncation handling by default whenever there's an explicit width restriction on the Badge: #68107.

I am assuming we never want multi-line badges. Let me know if you can think of use cases if this kind of auto-truncation would not work.

**Comment by jameskoster** [comment]
Now that you mention it, forcing truncation to avoid multi-line badges is probably a good guardrail to avoid misuse :) Min/max-width's may also be worth considering.

**Comment by tyxla** [comment]
I agree, and now that we have just created this component, it is a good time to impose that limitation.
